In [1]:
import pandas as pd


In [2]:
df=pd.read_csv("bollywood.csv")
df

SlNo Release Date                    MovieName ReleaseTime      Genre  \
0       1    18-Apr-14                     2 States          LW    Romance   
1       2     4-Jan-13                 Table No. 21           N  Thriller    
2       3    18-Jul-14           Amit Sahni Ki List           N     Comedy   
3       4     4-Jan-13             Rajdhani Express           N     Drama    
4       5     4-Jul-14                 Bobby Jasoos           N     Comedy   
..    ...          ...                          ...         ...        ...   
144   145    27-Feb-15           Dum Laga Ke Haisha           N     Comedy   
145   146    13-Mar-15                         NH10           N   Thriller   
146   147    20-Mar-15  Dilliwali Zaalim Girlfriend           N     Comedy   
147   148    20-Mar-15                     Hunterrr           N     Comedy   
148   149    23-May-14                Kochadaiiyaan          HS     Action   

     Budget  BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
0        36               104.00       8576361         26622             2527  
1        10                12.00       1087320          1129              137  
2        10                 4.00        572336           586               54  
3         7                 0.35         42626            86               19  
4        18                10.80       3113427          4512             1224  
..      ...                  ...           ...           ...              ...  
144      15                30.00       3250917          8185              615  
145      13                32.10       5592977         15464             1513  
146      32                12.00       2316047          4289              807  
147       5                11.89       4674795          3706              762  
148     150               120.00       4740727         13466             2649  

[149 rows x 10 columns]

### 1. How many records are present in the dataset? Print the metadata information of the dataset. 

In [4]:
df.shape[0]

149

In [5]:
df.info

<bound method DataFrame.info of      SlNo Release Date                    MovieName ReleaseTime      Genre  \
0       1    18-Apr-14                     2 States          LW    Romance   
1       2     4-Jan-13                 Table No. 21           N  Thriller    
2       3    18-Jul-14           Amit Sahni Ki List           N     Comedy   
3       4     4-Jan-13             Rajdhani Express           N     Drama    
4       5     4-Jul-14                 Bobby Jasoos           N     Comedy   
..    ...          ...                          ...         ...        ...   
144   145    27-Feb-15           Dum Laga Ke Haisha           N     Comedy   
145   146    13-Mar-15                         NH10           N   Thriller   
146   147    20-Mar-15  Dilliwali Zaalim Girlfriend           N     Comedy   
147   148    20-Mar-15                     Hunterrr           N     Comedy   
148   149    23-May-14                Kochadaiiyaan          HS     Action   

     Budget  BoxOfficeCollectio

     There are 149 records in the document of bollywood.and the information of metadata is above

### 2. How many movies got released in each genre? Which genre had highest number of releases? Sort number of releases in each genre in descending order

In [46]:
x = df['Genre'].value_counts()
x
x.sort_values(ascending = False)

Comedy       36
 Drama       35
Thriller     26
Romance      25
Action       21
Thriller      3
Action        3
Name: Genre, dtype: int64

    There are count of the each released movie of in each Genre.Here comedy Genre has highest no.of releases.And the descending order of no.of releases of each Genre is in the above..

### 3. How many movies in each genre got released in different release times like long weekend,festive season,etc..

In [57]:
df['ReleaseTime'].unique()

array(['LW', 'N', 'HS', 'FS'], dtype=object)

In [76]:
a = df[df['ReleaseTime'] == 'LW']['Genre'].value_counts()
b = df[df['ReleaseTime'] == 'N']['Genre'].value_counts()
c = df[df['ReleaseTime'] == 'HS']['Genre'].value_counts()
d = df[df['ReleaseTime'] == 'FS']['Genre'].value_counts()

print("no.of movies released in different release times is :",a)
print("no.of movies released in different release times is :",b)
print("no.of movies released in different release times is :",c)
print("no.of movies released in different release times is :",d)

no.of movies released in different release times is : Comedy       5
Romance      4
Action       3
Thriller     1
 Drama       1
Thriller     1
Name: Genre, dtype: int64
no.of movies released in different release times is :  Drama       24
Comedy       23
Thriller     20
Romance      15
Action       12
Action        3
Thriller      2
Name: Genre, dtype: int64
no.of movies released in different release times is :  Drama      6
Comedy      5
Romance     3
Action      3
Thriller    1
Name: Genre, dtype: int64
no.of movies released in different release times is :  Drama      4
Thriller    4
Action      3
Romance     3
Comedy      3
Name: Genre, dtype: int64


### 4. Which month of the year, maximum number movie releases are seen? (Extract month component from ReleaseDate column)

In [80]:
df['Release Date']

0      18-Apr-14
1       4-Jan-13
2      18-Jul-14
3       4-Jan-13
4       4-Jul-14
         ...    
144    27-Feb-15
145    13-Mar-15
146    20-Mar-15
147    20-Mar-15
148    23-May-14
Name: Release Date, Length: 149, dtype: object